src/libexpr/eval.cc

```diff
@@ -519,33 +519,40 @@ EvalState::EvalState(
     static_assert(sizeof(Env) <= 16, "environment must be <= 16 bytes");
 
     /* Initialise the Nix expression search path. */
-    evalSettings.nixPath.setDefault(evalSettings.getDefaultNixPath());
     if (!evalSettings.pureEval) {
```

Comment:
Is there anything speaking against

```cpp
if (evalSettings.pureEval) {
  createBaseEnv();
  return;
}
```
and then increasing looseness in the following?

The inverted logic makes reasoning quite cumbersome.